src/index.ml

```diff
@@ -464,9 +468,10 @@ struct
           Tbl.clear log.mem;
           iter_io add_log_entry log.io;
           may (fun (i : index) -> IO.close i.io) t.index;
-          if Int64.equal generation 0L then t.index <- None
+          let index_path = index_path t.root in
+          if Int64.equal generation 0L || not (Sys.file_exists index_path) then
+            t.index <- None
```

Comment:
Is this necessary now that the generation number is being incremented on clear?

Reply:
No, it seems not. (removed the test of generation)